node/src/accountant/scanners_tools.rs

```diff
@@ -0,0 +1,539 @@
+    }
+
+    #[test]
+    fn investigate_debt_extremes_picks_the_most_relevant_records() {
+        let now = SystemTime::now();
+        let now_t = to_time_t(now);
+        let same_amount_significance = 2_000_000;
+        let same_age_significance = from_time_t(now_t - 30000);
+        let payables = &[
+            PayableAccount {
+                wallet: make_wallet("wallet0"),
+                balance: same_amount_significance,
+                last_paid_timestamp: from_time_t(now_t - 5000),
+                pending_payable_opt: None,
+            },
+            //this debt is more significant because beside being high in amount it's also older, so should be prioritized and picked
+            PayableAccount {
+                wallet: make_wallet("wallet1"),
+                balance: same_amount_significance,
+                last_paid_timestamp: from_time_t(now_t - 10000),
+                pending_payable_opt: None,
+            },
+            //similarly these two wallets have debts equally old but the second has a bigger balance and should be chosen
+            PayableAccount {
+                wallet: make_wallet("wallet3"),
+                balance: 100,
+                last_paid_timestamp: same_age_significance,
+                pending_payable_opt: None,
+            },
+            PayableAccount {
+                wallet: make_wallet("wallet2"),
+                balance: 330,
+                last_paid_timestamp: same_age_significance,
+                pending_payable_opt: None,
+            },
+        ];
+
+        let result = investigate_debt_extremes(now, payables);
+
+        assert_eq!(result, "Payable scan found 4 debts; the biggest is 2000000 owed for 10000sec, the oldest is 330 owed for 30000sec")
+    }
+
+    #[test]
+    fn balance_and_age_is_calculated_as_expected() {
+        let now = SystemTime::now();
+        let offset = 1000;
+        let receivable_account = ReceivableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: 10_000_000_000,
+            last_received_timestamp: from_time_t(to_time_t(now) - offset),
+        };
+
+        let (balance, age) = balance_and_age(now, &receivable_account);
+
+        assert_eq!(balance, "10");
+        assert_eq!(age.as_secs(), offset as u64);
+    }
+
+    #[test]
+    fn separate_early_errors_works() {
+        let payable_ok = Payable {
+            to: make_wallet("blah"),
+            amount: 5555,
+            timestamp: SystemTime::now(),
+            tx_hash: Default::default(),
+        };
+        let error = BlockchainError::SignedValueConversion(666);
+        let sent_payable = SentPayable {
+            timestamp: SystemTime::now(),
+            payable: vec![Ok(payable_ok.clone()), Err(error.clone())],
+            response_skeleton_opt: None,
+        };
+
+        let (ok, err) = separate_early_errors(&sent_payable, &Logger::new("test"));
+
+        assert_eq!(ok, vec![payable_ok]);
+        assert_eq!(err, vec![error])
```

Comment:
Consider to add a log assertion to the end; even though it had been a part of it when you manipulated it.

Not necessarily mandatory though.